include/mbgl/platform/default/headless_view.hpp

```diff
@@ -66,25 +65,24 @@ class HeadlessView : public View {
 
     bool needsResize = false;
     bool extensionsLoaded = false;
-    bool active = false;
 
 #if MBGL_USE_QT
-    QGLWidget* glContext = nullptr;
+    static QGLWidget* glContext;
 #endif
 
 #if MBGL_USE_CGL
-    CGLContextObj glContext = nullptr;
+    static CGLContextObj glContext;
 #endif
 
 #if MBGL_USE_EAGL
-    void *glContext = nullptr;
+    static void *glContext;
 #endif
 
 #if MBGL_USE_GLX
-    Display *xDisplay = nullptr;
-    GLXFBConfig *fbConfigs = nullptr;
-    GLXContext glContext = nullptr;
-    GLXPbuffer glxPbuffer = 0;
+    static Display *xDisplay;
```

Comment:
Capturing from our chat, these `static` are probable why it is crashing on GLFW.